src/ja/common/work_machine.py

```diff
@@ -15,28 +15,86 @@ def __init__(self, cpu_threads: int, memory: int, swap: int):
         @param memory The amount of RAM, in MB.
         @param swap The amount of swap space, in MB.
         """
+        if cpu_threads < 0:
+            raise ValueError("CPU threads cannot be negative")
+        if memory < 0:
+            raise ValueError("Memory cannot be negative")
+        if swap < 0:
+            raise ValueError("Swap cannot be negative")
```

Comment:
What is the purpose of these checks?
Judging by the code you added below, any of these properties can become negative, correct?

Reply:
Correct, I made a mistake, those properties can be negative. But cpu_threads and so on can't, so I'll move those checks to the WorkMachineResources class.

Reply:
Hmm, now that I think of it, I don't think it makes sense for a negative resource allocation??

Reply:
It doesn't, but it's useful. I updated the code, I moved those checks to WorkMachineResources in database.types.work_machine.
So now, a ResourceAllocation can have negative attributes but a WorkMachineResources instance can't.